apps/web/src/hooks/useIsContract.ts

```diff
@@ -0,0 +1,18 @@
+import { ethers } from 'ethers'
+import useSWRImmutable from 'swr/immutable'
+
+import { RPC_URL } from 'src/constants/rpc'
+import { AddressType, CHAIN_ID } from 'src/typings'
+
+export const useIsContract = ({
+  address,
+  chainId = CHAIN_ID.ETHEREUM,
+}: {
+  address?: AddressType
+  chainId?: CHAIN_ID
+}) => {
+  return useSWRImmutable(address ? [address, chainId] : undefined, async (address) => {
+    const provider = new ethers.providers.JsonRpcProvider(RPC_URL[chainId])
+    return await provider.getCode(address).then((x) => x !== '0x')
```

Comment:
would this by chance be the equivalent here? just wondering if we could instead use viem. do we generally still have a dependency on ethers?

Reply:
https://viem.sh/docs/contract/getBytecode.html#json-rpc-method

Reply:
migrating to viem and removing ethers is dependent on #350 so once that is in we will start replacing all ethers dependencies. for now sticking with ethers